test/model/sequence_result_test.rb

```diff
@@ -3,11 +3,310 @@
+
+    assert_equal Inferno::ResultStatuses::PASS, sequence_result.result
+  end
+
+  def test_sequence_result_fail
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = true
+    test_result2.result = Inferno::ResultStatuses::SKIP
+    sequence_result.test_results << test_result2
+
+    test_result3 = Inferno::Models::TestResult.new
+    test_result3.required = true
+    test_result3.result = Inferno::ResultStatuses::FAIL
+    sequence_result.test_results << test_result3
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::FAIL, sequence_result.result
+  end
+
+  def test_sequence_result_pass_when_fail_optional
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = false
+    test_result2.result = Inferno::ResultStatuses::FAIL
+    sequence_result.test_results << test_result2
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::PASS, sequence_result.result
+  end
+
+  def test_sequence_result_error
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = true
+    test_result2.result = Inferno::ResultStatuses::SKIP
+    sequence_result.test_results << test_result2
+
+    test_result3 = Inferno::Models::TestResult.new
+    test_result3.required = true
+    test_result3.result = Inferno::ResultStatuses::ERROR
+    sequence_result.test_results << test_result3
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::ERROR, sequence_result.result
+  end
+
+  def test_sequence_result_pass_when_error_optional
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = false
+    test_result2.result = Inferno::ResultStatuses::ERROR
+    sequence_result.test_results << test_result2
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::PASS, sequence_result.result
+  end
+
+  def test_sequence_result_pass_fail_vs_error
```

Comment:
This function name is really confusing. What should be the result? Is it pass, or fail or error?

Reply:
renamed test to test_sequence_result_error_when_pass_fail_and_error_tests_in_sequence